### PR TITLE
[SPARK-49971][SQL] Assign proper error condition for _LEGACY_ERROR_TEMP_1097

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign proper error condition & sqlstate for` _LEGACY_ERROR_TEMP_1097`

### Why are the changes needed?

To improve the error message by assigning proper error condition and SQLSTATE

### Does this PR introduce _any_ user-facing change?

No, only user-facing error message improved

### How was this patch tested?

Updated the existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -4852,6 +4852,11 @@
     ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_DATA_TYPE_FOR_CORRUPTED_RECORD" : {
+    "message" : [
+      "The field <columnName> for corrupt records must be string type and nullable."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:4857]:
```suggestion
      "The column <columnName> for corrupt records must have the nullable STRING type, but got <actualType>"
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 2]
```diff
@@ -4852,6 +4852,11 @@
     ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_DATA_TYPE_FOR_CORRUPTED_RECORD" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:4855]:
```suggestion
  "INVALID_CORRUPT_RECORD_TYPE" : {
```

File: common/utils/src/main/resources/error/error-conditions.json [hunk 3]
```diff
@@ -4852,6 +4852,11 @@
...
+    "message" : [
+      "The field <columnName> for corrupt records must be string type and nullable."
+    ]
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:4858]: please, set `sqlState`

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by itholic** [comment]
Thanks @MaxGekk for the review! Applied the comments

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
@itholic Could you fix the failed tests:
```
[info] - verify corrupt column *** FAILED *** (3 milliseconds)
[info]   (non-codegen mode) Expected error message is `The field for corrupt records must be string type and nullable`, but `[INVALID_CORRUPT_RECORD_TYPE] The column `_unparsed` for corrupt records must have the nullable STRING type, but got "BOOLEAN". SQLSTATE: 42804` found (ExpressionEvalHelper.scala:226)
```

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**